invokeai/app/api/routers/model_manager.py

```diff
@@ -816,3 +814,61 @@ async def get_starter_models() -> list[StarterModel]:
         model.dependencies = missing_deps
 
     return starter_models
+
+
+@model_manager_router.get(
+    "/model_cache",
+    operation_id="get_cache_size",
+    response_model=float,
+    summary="Get maximum size of model manager RAM or VRAM cache.",
+)
+async def get_cache_size(cache_type: CacheType = Query(description="The cache type", default=CacheType.RAM)) -> float:
+    """Return the current RAM or VRAM cache size setting (in GB)."""
+    cache = ApiDependencies.invoker.services.model_manager.load.ram_cache
+    return cache.max_cache_size if cache_type == CacheType.RAM else cache.max_vram_cache_size
+
+
+@model_manager_router.put(
+    "/model_cache",
+    operation_id="set_cache_size",
+    response_model=float,
+    summary="Set maximum size of model manager RAM or VRAM cache, optionally writing new value out to invokeai.yaml config file.",
+)
+async def set_cache_size(
+    value: float = Query(description="The new value for the maximum cache size"),
+    cache_type: CacheType = Query(description="The cache type", default=CacheType.RAM),
+    persist: bool = Query(description="Write new value out to invokeai.yaml", default=False),
+) -> float:
+    """Set the current RAM or VRAM cache size setting (in GB). ."""
+    cache = ApiDependencies.invoker.services.model_manager.load.ram_cache
+    app_config = get_config()
+    if cache_type == CacheType.RAM:
+        cache.max_cache_size = value
+        app_config.ram = value
+    elif cache_type == CacheType.VRAM:
+        cache.max_vram_cache_size = value
+        app_config.vram = value
+
+    if persist:
+        config_path = app_config.config_file_path
+        print(f"DEBUG: config_path = {config_path}")
+        try:
+            shutil.copy(config_path, config_path.with_suffix(".yaml.bak"))
+            app_config.write_file(config_path)
+        except Exception as e:
+            shutil.move(config_path.with_suffix(".yaml.bak"), config_path)
+            raise RuntimeError(f"Failed to write modified configuration to {config_path}: {e}") from e
+
+    return cache.max_vram_cache_size if cache_type == CacheType.VRAM else cache.max_cache_size
```

Comment:
Same comment as earlier about explicitly handling all enum values and raising in `else`.

Reply:
Fixed.